### PR TITLE
Microsoft.Azure.Management.Avs 3.0.0

### PR DESCRIPTION
**Pull request by cataggar**
This updates the Microsoft.Azure.Management.Avs SDK to use the 2021-12-01 API from https://github.com/Azure/azure-rest-api-specs/pull/16426.

It will be a major version update to 3.0.0 to be published to:
https://www.nuget.org/packages/Microsoft.Azure.Management.Avs/

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [x] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by cataggar** [comment]
The [guide for publishing](https://github.com/Azure/azure-sdk-for-net/pull/12602#issuecomment-647841643) to nuget.org is gone. What is the new process?

**Comment by cataggar** [comment]
I had to specify the `main` branch. #25292

File: sdk/avs/Microsoft.Azure.Management.Avs/tests/AvsTests.cs [hunk 1]
```diff
@@ -126,22 +97,13 @@ public void PasswordResets()
...
-
-            avsClient.PrivateClouds.RotateNsxtPassword(rgName, cloudName);
-            avsClient.PrivateClouds.RotateVcenterPassword(rgName, cloudName);
+            using var avsClient = context.GetServiceClient<AvsClient>();
 
-            var credsC = avsClient.PrivateClouds.ListAdminCredentials(rgName, cloudName);
-            Assert.NotEqual(credsA.NsxtPassword, credsC.NsxtPassword);
-            Assert.NotEqual(credsA.VcenterPassword, credsC.VcenterPassword);
+            avsClient.PrivateClouds.ListAdminCredentials(rgName, cloudName);
+            // TODO replace LRO responses in test server
```
Comment (cataggar) [inline sdk/avs/Microsoft.Azure.Management.Avs/tests/AvsTests.cs:104]: @Pinak-cb, I fixed the test server (mock_api). Please reenable these two tests.

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure VMware Solution Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure VMware Solution.")]
 
-[assembly: AssemblyVersion("2.0.0")]
-[assembly: AssemblyFileVersion("2.0.0.1")]
+[assembly: AssemblyVersion("3.0.0")]
+[assembly: AssemblyFileVersion("3.0.0.1")]
```
Comment (archerzz) [inline sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs:11]: I think you can change the `AssemblyFileVersoin` to `3.0.0.0`

**Review by archerzz — APPROVED** [review]

**Comment by archerzz** [comment]
@cataggar Looks good to me. If you still choose `3.0.0.1` as the assembly file version, then no change is required, and we can merge this PR. Thanks.

File: sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (cataggar) [reply sdk/avs/Microsoft.Azure.Management.Avs/src/Properties/AssemblyInfo.cs:11]: I'll leave as `3.0.0`. Sounds like that will work.

**Comment by cataggar** [comment]
Thanks for approving @archerzz, can you please merge this? How do we get it published to nuget.org?

**Comment by archerzz** [comment]
@cataggar Merging. You can refer to this link regarding how to publish to nuget.org: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges

**Pull request merged**